### DISCUSSION TITLE
xterm behavior

### DISCUSSION THREAD

**Discussion by pmassey** [iraf-community/iraf#334] [Q&A]
Hi Ole--Sorry to bother you again.  I've installed the community version of IRAF and X11iraf on my new M2 MacBook Air running Ventura.  Most things are working great.  But, I'm noticing a problem. 

First, xgterm doesn't seem to recognized the delete key.   It seems to map to a ^H.  (This is with or without the cl running.). Connected with this is that one can't go back up with "eh" and usefully edit a prior command.  I don't know if you've seen this.

Thanks!
Best regards,
Phil

**Comment by olebole** [comment]
Unfortunately I don't own a Mac, so it is hard to reproduce for me. If you use zsh or tcsh, you could try this tip:
https://sparky.rice.edu/~hartigan/del.html
For bash, the equivalent command (but with different syntax) is "bind".
A generic input configuration for readline (and probably libedit, which is used on Macs) would be `/etc/inputrc` or `~/.inputrc/` or `~/.editrc`, where one can set the key bindings.

If you have success, please report it back. As I don't think it is a bug of xgterm, I will move it to the discussion page.

**Comment by pmassey** [comment]
Sadly, the tip did not work for me.  I’ll note that the problem doesn’t happen in a regular xterm window: the delete key works just fine.  It’s only within an xgterm that I see this behavior.

It also seems as my .tcshrc is not being invoked when I open up an xgterm.
(But I explicitly source it to see if this solved the problem.) I think this is normal behavior, though, isn’t it?

Thanks for any additional comments or suggestions!

**Reply by pmassey** [reply]
I've achieved partial success.  If I insert "!stty erase ^H" into the login.cl as Hartigan suggests, delete works fine in the cl on the line you're typing.  You still can't edit a previous line by going back with an "eh" and "verify" turned on. This is a big help.

**Comment by olebole** [comment]
One additional question: Did you actually run `tic xgterm/xgterm.terminfo` to compile the terminfo entry for xgterm?

**Reply by pmassey** [reply]
No, I didn't.  Where do I need to be in order to run it?  If I just enter tic xgterm/xgterm.terminfo
I get:
tic: xgterm/xgterm.terminfo No such file or directory
Thanks!

**Reply by olebole** [reply]
This should be entered from the base directory of the x11iraf sources. Or at least the command line argument should point to `xgterm/xgterm.terminfo` in the x11iraf sources.

**Comment by pmassey** [comment]
Hi Ole---It's been a few months since I installed X11IRA,  so my memory was a little vague about what I did.  So, I tried re-installing it today afresh.  "make" runs successfully; however, when I run sudo install make, I get a segmentation error as soon as it hits the tic xgterm/xgterm.terminfo command. Running tic xgterm/xgterm.terminfo from the command line has the same effect.

Just as a reminder, I'm running this on an Apple Silicon (M2) MacBook Air using Ventura OS (13.5)

Here's the screen dump:
```
sudo make install
Password:
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C xgterm
make[1]: Nothing to be done for `all'.
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C ximtool
cd clients && mkpkg || true
hlib$mkpkg.inc:5: warning: macro `IRAFARCH' not found
$call relink
$call libpkg.a
mkpkg[1]: Entering directory `/Users/massey/x11iraf-2.1/ximtool/clients/lib/'
Subdirectory lib is up to date
mkpkg[1]: Leaving directory `/Users/massey/x11iraf-2.1/ximtool/clients/lib/'
mkpkg[1]: Entering directory `/Users/massey/x11iraf-2.1/ximtool/clients/wcspix/'
Subdirectory wcspix is up to date
mkpkg[1]: Leaving directory `/Users/massey/x11iraf-2.1/ximtool/clients/wcspix/'
Library libpkg.a is up to date
Object x_ism.o is up to date
/usr/local/lib/iraf/unix/bin/xc.e -Nz   -z x_ism.o libpkg.a -o ism_wcspix.e -lds -lxtools -liminterp -lslalib
mkdir -p /usr/local/bin /usr/local/man/man1
install -m755 xgterm/xgterm /usr/local/bin
install -m755 xgterm/xgterm.man /usr/local/man/man1/xgterm.1
tic xgterm/xgterm.terminfo
make: *** [install] Segmentation fault: 11
```
And similarly, 
```
tic xgterm/xgterm.terminfo
Segmentation fault
```

**Reply by pmassey** [reply]
Actually I was able to get around this.  The version of tic that was being used was from anaconda; when I ran the /usr/bin/tic version it ran fine.  That said, it doesn't seem to have made any difference.
